src/main/scala/io/spicelabs/goatrodeo/util/Syft.scala

```diff
@@ -0,0 +1,261 @@
+package io.spicelabs.goatrodeo.util
+
+import com.typesafe.scalalogging.Logger
+import org.json4s.*
+import org.json4s.native.JsonMethods.*
+
+import java.io.ByteArrayOutputStream
+import java.nio.file.Path
+import scala.util.Try
+import io.spicelabs.goatrodeo.omnibor.StringOrPair
+import io.spicelabs.goatrodeo.omnibor.ConnectionAugmentation
+import io.spicelabs.goatrodeo.omnibor.EdgeType
+import io.spicelabs.goatrodeo.omnibor.*
+
+/** The bridge to Syft https://github.com/anchore/syft
+  */
+object Syft {
+  private lazy val syftMimeTypes = Set(
+    "application/zip",
+    "application/java-archive",
+    "application/vnd.android.package-archive",
+    "application/x-rpm",
+    "application/x-archive",
+    "application/x-iso9660-image",
+    "application/x-gtar",
+    "application/x-debian-package",
+    "application/gzip",
+    "application/zstd"
+  )
+
+  /** Is the artifact a container that Syft knowns how to process?
+    *
+    * @param artifact
+    * @return
+    */
+  def isContainer(artifact: ArtifactWrapper): Boolean = {
+    syftMimeTypes.contains(artifact.mimeType)
+  }
+
+  lazy val hasSyft: Boolean = {
+    Try {
+      val process = ProcessBuilder("syft", "--help").start()
+      while (process.isAlive()) {
+        Thread.sleep(100)
+      }
+      process.exitValue() == 0
+    }.toOption == Some(true)
+  }
+
+  def runSyftFor(
+      artfiact: ArtifactWrapper,
+      tempDir: Path
+  ): Option[SyftResult] = {
+    if (hasSyft) {
+      val targetFile = artfiact.forceFile(tempDir)
+      val pb = ProcessBuilder(
+        List(
+          "syft",
+          "scan",
+          f"file:${targetFile.getName()}",
+          "--output",
+          "syft-json"
+        )*
```

Comment:
Are both "Syft failed with an exit code > 0" and "syft not found" handled here? (not sure about the underlying system but that tends to be two cases that need to be handled separately)

Reply:
Syft not found is handled here: https://github.com/spice-labs-inc/goatrodeo/pull/133/files#diff-226c51ef1558ee91e2f1ae6d5b9dfe05d62bfc6b385ddfd003b25db8650243f5R40-R48

Also, because it's in a `Try` block, if it fails (not returning != 0, but something else worse goes wrong) the `Process` code will throw an exception.

Net-net, if it doesn't run correctly, it will simply not add metadata to the ADG

Reply:
Good good. (not found can still happen because it can be deleted mid-run in theory. Not that one should, but races are in mind.)